src/main/kotlin/org/rust/cargo/toolchain/tools/Cargo.kt

```diff
@@ -391,6 +391,9 @@ class Cargo(
                     }
                 }
                 add(command)
+                if(command == "vendor"){
+                    add("--respect-source-config")
+                }
```

Comment:
`toGeneralCommandLine` is a too generic method, and I'd prefer to avoid any hacks for particular commands here.
Since we need to pass `--respect-source-config` only for `cargo vendor` command and we use `cargo vendor` in single place, let's just pass this additional flag in `vendorDependencies` method

Reply:
thanks, i have done it ,please check it again